### PR TITLE
Add pytest suite for airflow processor

### PR DESCRIPTION
**Pull request by akchinSTC**
See original PR #490 for more details
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by akchinSTC** [comment]
depends and based on #1316

File: elyra/pipeline/tests/test_airlfow_processor.py
Comment (kevin-bates) [inline elyra/pipeline/tests/test_airlfow_processor.py:45]: I don't know if this is problematic, but I think we should avoid prefixing fixture names with `test_` since those methods _are_ the tests and part of the collection process (with fixtures occurring first I guess).  Perhaps `af_metadata` or `airflow_metadata`.

File: elyra/pipeline/tests/test_airlfow_processor.py
Comment (kevin-bates) [inline elyra/pipeline/tests/test_airlfow_processor.py:60]: Same comment as `test_metadata`.

**Review by kevin-bates — COMMENTED** [review]
Thanks for putting this together.  I just had a comment on the fixture names.

Also, there's a typo in the test filename: `test_airlfow_processory.py`

File: elyra/pipeline/processor_airflow.py
Comment (kevin-bates) [inline elyra/pipeline/processor_airflow.py:76]: Could we collect the metadata instance value into a `github_api_endpoint` variable - when the other github-related values are collected, then reference that variable here?

Comment (akchinSTC) [reply elyra/pipeline/processor_airflow.py]: check and done.

File: elyra/pipeline/tests/test_airlfow_processor.py
Comment (ptitzler) [inline elyra/pipeline/tests/test_airlfow_processor.py:1]: The file name contains a typo

**Comment by ptitzler** [comment]
Will a github method test be added separately?

File: elyra/pipeline/tests/test_airlfow_processor.py
Comment (ptitzler) [reply elyra/pipeline/tests/test_airlfow_processor.py:1]: Too bad it wasn't an airfowl (:

Comment (akchinSTC) [reply elyra/pipeline/tests/test_airlfow_processor.py:1]: ouch..lol, fixed

**Comment by akchinSTC** [comment]
@ptitzler -yes, will add tests for the git utils, but can go separately in another PR

**Comment by lresende** [comment]
@ptitzler and @kevin-bates It seems that you guys have been reviewing this, so I will wait for your final approvals before merging.

**Review by ptitzler — APPROVED** [review]
LGTM

File: elyra/pipeline/tests/test_airflow_processor.py
Comment (kevin-bates) [inline elyra/pipeline/tests/test_airflow_processor.py:139]: Could you add a comment indicating how this hash value is computed so folks making changes know what to do?

Frankly. I'd prefer we validate the export w/o this dependency, perhaps by looking for expected values in the output python file - or things like that.  This hash stuff has proven to be quite fragile and was abandoned in [kfp-notebook](https://github.com/elyra-ai/kfp-notebook/pull/68).

File: elyra/pipeline/tests/test_airflow_processor.py
Comment (kevin-bates) [inline elyra/pipeline/tests/test_airflow_processor.py:193]: This test feels more appropriate as a parser-based test.  Is it unique to Airflow?

(Btw, I pity the person that needs to update the test pipeline file someday!)

File: elyra/pipeline/tests/test_airflow_processor.py
Comment (akchinSTC) [reply elyra/pipeline/tests/test_airflow_processor.py:139]: +1 abandoning the hash stuff. Ill refactor to verify via expected values

File: elyra/pipeline/tests/test_airflow_processor.py
Comment (akchinSTC) [reply elyra/pipeline/tests/test_airflow_processor.py:193]: the ordered dictionary is unique to the airflow processor. the kfp processor just creates a dictionary of notebook operations. The order is important in that its needed to correctly build the DAG template.

**Comment by akchinSTC** [comment]
Im working through parsing the generated airflow DAG file. Will have an updated PR no later than EOD today. Attempted to use the ast lib to parse but that felt heavy.

**Review by kevin-bates — APPROVED** [review]
LGTM - took tests for a spin - all good.  Thanks for removing the hash check - this will be much more sustainable.

**Pull request merged**
